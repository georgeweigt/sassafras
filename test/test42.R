A = c(6,0,2,8,11,4,13,1,8,0)
B = c(0,2,3,1,18,4,14,9,1,9)
C = c(13,10,18,5,23,12,5,16,1,20)

print(t.test(A,B,var.equal=TRUE))
print(t.test(A,C,var.equal=TRUE))
print(t.test(B,C,var.equal=TRUE))

YA = c(6,0,2,8,11,4,13,1,8,0)
YB = c(0,2,3,1,18,4,14,9,1,9)
sse = var(YA) * (length(YA) - 1) + var(YB) * (length(YB) - 1)
dfe = length(YA) + length(YB) - 2
mse = sse / dfe
se = sqrt(mse * (1 / length(YA) + 1 / length(YB)))
t = (mean(YA) - mean(YB)) / se
mean(YA) - mean(YB) - qt(0.975,dfe) * se
mean(YA) - mean(YB) + qt(0.975,dfe) * se

print(mean(YA) - mean(YB) - qt(0.975,dfe) * se)
print(mean(YA) - mean(YB) + qt(0.975,dfe) * se)

print(2 * (1 - pt(t,dfe)))

print(t.test(A,B,var.equal=TRUE))
