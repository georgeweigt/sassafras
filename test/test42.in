data
input trt$ y @@
datalines
A 6    A 0    A 2    A 8   A 11
A 4    A 13   A 1    A 8   A 0
B 0    B 2    B 3    B 1   B 18
B 4    B 14   B 9    B 1   B 9
C 13   C 10   C 18   C 5   C 23
C 12   C 5    C 16   C 1   C 20
;
proc anova
model y = trt
means trt / lsd ttest

