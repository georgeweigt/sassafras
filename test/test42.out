                              Analysis of Variance

    Source     DF     Sum of Squares      Mean Square     F Value     Pr > F
    Model       2       293.60000000     146.80000000        3.98     0.0305
    Error      27       995.10000000      36.85555556                       
    Total      29      1288.70000000                                        

                R-Square     Coeff Var     Root MSE       Y Mean
                0.227826     76.846553     6.070878     7.900000

     Source     DF         Anova SS      Mean Square     F Value     Pr > F
     TRT         2     293.60000000     146.80000000        3.98     0.0305

                                 Mean Response

             TRT      N        Mean Y     95% CI MIN     95% CI MAX
             A       10      5.300000       1.360938       9.239062
             B       10      6.100000       2.160938      10.039062
             C       10     12.300000       8.360938      16.239062

                       Least Significant Difference Test

  TRT    TRT      Delta Y    95% CI MIN    95% CI MAX    t Value    Pr > |t|  
  A      B      -0.800000     -6.370676      4.770676      -0.29      0.7705  
  A      C      -7.000000    -12.570676     -1.429324      -2.58      0.0157 *
  B      A       0.800000     -4.770676      6.370676       0.29      0.7705  
  B      C      -6.200000    -11.770676     -0.629324      -2.28      0.0305 *
  C      A       7.000000      1.429324     12.570676       2.58      0.0157 *
  C      B       6.200000      0.629324     11.770676       2.28      0.0305 *

                               Two Sample t-Test

  TRT    TRT      Delta Y    95% CI MIN    95% CI MAX    t Value    Pr > |t|  
  A      B      -0.800000     -5.922306      4.322306      -0.33      0.7466  
  A      C      -7.000000    -12.664270     -1.335730      -2.60      0.0182 *
  B      A       0.800000     -4.322306      5.922306       0.33      0.7466  
  B      C      -6.200000    -12.467653      0.067653      -2.08      0.0523  
  C      A       7.000000      1.335730     12.664270       2.60      0.0182 *
  C      B       6.200000     -0.067653     12.467653       2.08      0.0523  

